cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+                    cy.get('[aria-label="Download"]').click({ force: false });
+
+                    // Assert that the file was downloaded
+                    cy.readFile("cypress/downloads/1sec.ogg").should("exist");
+                });
+        });
+    });
+
+    it("should support replying to audio file with another audio file", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Assert the audio player is rendered
+            cy.get(".mx_EventTile_last .mx_AudioPlayer_container").should("exist");
+
+            cy.get(".mx_EventTile_last")
+                .realHover()
+                .within(() => {
+                    // Click "Reply" button on MessageActionBar
+                    cy.get('[aria-label="Reply"]').click({ force: false });
+                });
+        });
+
+        // Reply to the player with another audio file
+        uploadFile("cypress/fixtures/1sec.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            cy.get(".mx_EventTile_last").within(() => {
+                // Assert that replied audio file is rendered as file button inside ReplyChain
+                cy.get(".mx_ReplyChain_wrapper").within(() => {
+                    cy.get(".mx_MFileBody_info[role='button']").within(() => {
+                        // Assert that the file button has file name
+                        cy.get(".mx_MFileBody_info_filename").should("exist");
+                    });
+                });
+
+                cy.get(".mx_MAudioBody").within(() => {
+                    // Assert that the play button is visible
+                    cy.get("[data-testid='play-pause-button'][aria-label='Play']").should("be.visible");
+                });
+            });
+        });
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            const takeSnapshotReply = (layout: string) => {
+                // Click the timestamp to highlight the event tile on which player is rendered
+                cy.get(".mx_EventTile_last .mx_MessageTimestamp").click();
+
+                cy.get(".mx_EventTile_last.mx_EventTile_selected").within(() => {
+                    checkPlayerVisibility();
+                });
+
+                cy.get(".mx_EventTile_last").percySnapshotElement(
+                    `Selected EventTile of audio player with a reply on ${layout} layout`,
+                    {
+                        percyCSS,
+                        widths: snapshotWidthsEventTile,
+                    },
+                );
+                cy.log(`Took a snapshot of selected EventTile of audio player with a reply on ${layout} layout`);
+            };
```

Comment:
It feels like we have code very similar to this four times (twice in `takeSnapshots`, once here, once in `takeSnapshotReplyChain`). In fact this is identical to `takeSnapshotReplyChain`. Maybe we could factor out a common function?

It looks like the main difference with `takeSnapshots` is about whether we take a snapshot of the `EventTile` or just the AudioBody. Is that difference important, or could we do the same in both cases?

Reply:
I feared that adding a common function would complicated the PR. If it is fine, I'll add something like `takeSnapshot(layout, object)`, for example `takeSnapshot("IRC", "mx_EventTile_last")` or `takeSnapshot("bubble", "mx_AudioBody")`. This indeed seems to be much more simple.

> whether we take a snapshot of the `EventTile` or just the AudioBody. Is that difference important, or could we do the same in both cases?

I believe the difference would become crucial when we troubleshoot. Checking both (the audio player itself and `mx_EventTile` on which the player is implemented and rendered) should help us to understand whether a regression is due to the player itself, `mx_EventTile`, or both of them.

Reply:
> > whether we take a snapshot of the `EventTile` or just the AudioBody. Is that difference important, or could we do the same in both cases?
> 
> I believe the difference would become crucial when we troubleshoot. Checking both (the audio player itself, and `mx_EventTile` on which the player is implemented and rendered) should help us to understand whether a regression is due to the player itself, `mx_EventTile`, or both of them.

Hrm, I'm not convinced.

If a regression occurs, I think it will be easy to see if the problem is due to the event tile or the player; if necessary, we can use `git bisect`. On the other hand, by taking snapshots of the audio player:
 * it means there are differences between `takeSnapshotReply` and `takeSnapshots`, which means more code to understand and maintain
 * we have to carefully calculate the width of the audio player for the snapshot (and keep it in step with the design).

I think it would be better to always take a snapshot of the event tile.

Reply:
I agree with you. I have been quite hysterical about the strictness which I thought would be required for checking the players (voice and audio) :-D